runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -501,17 +502,16 @@ ClientSessionData::clearCaches()
 
    _classChainDataMap.clear();
 
-   // Free CHTable 
-   for (auto& it : _chTableClassMap)
-      {
-      TR_PersistentClassInfo *classInfo = it.second;
-      classInfo->removeSubClasses();
-      jitPersistentFree(classInfo);
-      }
-   _chTableClassMap.clear();
    _registeredJ2IThunksMap.clear();
    _registeredInvokeExactJ2IThunksSet.clear();
    _bClassUnloadingAttempt = false;
+
+   if (_chTable)
+      {
+      // Free CH table
+      _chTable->~JITServerPersistentCHTable();
+      jitPersistentFree(_chTable);
+      }
```

Comment:
We need to set the _chTable to NULL after freeing it. The session may continue to live even after caches are cleared.

Reply:
Fixed